addons/udes_warehouse_classification/models/stock_picking.py

```diff
@@ -6,16 +6,24 @@ class StockPicking(models.Model):
     _inherit = "stock.picking"
 
     def _get_classification_messages_for_product_picking(self):
-        """Return the product with classification messages appropriate for a picking"""
+        """
+        Return all products with classification messages appropriate to warn users. It is getting
+        all products, not only the products of the picking because it may be some products with
+        classifications that are not inside the picking but we still need to warn users.
+        """
+        Product = self.env["product.product"]
+        self.ensure_one()
         product_classifications = super()._get_classification_messages_for_product_picking()
-        products = self.mapped("move_lines.product_id")
+        products = Product.search([("u_has_warehouse_classifications", "=", True)])
```

Comment:
Also do we need the extra field `u_has_warehouse_classifications`? Could we perhaps not just have something like 
```suggestion
        products = Product.search([("u_product_warehouse_classification_ids", "!=", False)])
```

Reply:
I think that i tried , but didnt work. And thought is safer to check with a boolean field. Good spot though

Reply:
I was also thinking that, I think SQL can handle that so Odoo should be able too? Let's see.

Reply:
sorry I meant 
```suggestion
        products = Product.search([]).filtered(lamdba p: p.u_product_warehouse_cassification_ids)
```

Reply:
I am not sure because the new field does also have a compute method which goes through every product in self anytime a new u_product_warehouse_classification_ids is added to a product, which is a lot of iterations.

Reply:
Yes but u_product_warehouse_classification_ids are not added very frequently , and only added on form view which will be only one product.
While Goods in or other flows that use the same controller are used a lot, which trigger calling the method.
I think is better to ask a TL about this if you dont agree

Reply:
Yeah I see your point, it would be worse if the user on the HHT is slowed down during Goods in then the desktop user being slowed down in form view.

Reply:
Expect of types of user, u_product_warehouse_classification_ids is a config field which we think will be used less frequently than operations.
I mean its not that we give priority to users on HHT comparing to the desktop users.